file_reader.go

```diff
@@ -38,11 +49,25 @@ func (c *Client) Open(name string) (*FileReader, error) {
 		return nil, &os.PathError{"open", name, interpretException(err)}
 	}
 
+	status, ok := info.Sys().(*FileStatus)
+	if !ok {
+		return nil, &os.PathError{"open", name, errors.New("internal error: fail to access file status")}
+	}
+
+	var enc *transparentEncryptionInfo
+	if status.FileEncryptionInfo != nil {
+		enc, err = c.kmsGetKey(status.FileEncryptionInfo)
+		if err != nil {
+			return nil, &os.PathError{"open", name, err}
+		}
+	}
+
 	return &FileReader{
 		client: c,
 		name:   name,
 		info:   info,
 		closed: false,
+		enc:    enc,
```

Comment:
Looking at the `cipher` api a bit more, it seems to me there's a way cleaner to do this.

 - set `f.cipher` to a `cipher.Stream` with the key
 - in `Read`, deep in the loop, do 

```go
if (f.cipher != nil) { 
    n, err = cipher.StreamReader{S: f.cipher, R: f.blockReader}.Read(b) 
} else { 
    n, err = f.blockReader.Read(b)
}
```

That should be all that's needed.